src/openvic-simulation/types/Signal.hpp

```diff
@@ -506,9 +506,28 @@ namespace OpenVic::_detail::signal {
 
 	template<typename Lockable>
 	struct basic_observer : private observer_type {
-		virtual ~basic_observer() = default;
+	private:
+		Lockable mutex{};
+		memory::vector<scoped_connection> connections;
+
+		template<typename, typename...>
+		friend class basic_signal;
+		void add_connection(connection conn) {
+			std::unique_lock<Lockable> _ { mutex };
+			connections.emplace_back(std::move(conn));
+		}
```

Comment:
This should not be moved here, there is no excuse, I don't care what the styleguide says, its wrong and useless, don't move things simply to comply with it.